### PR TITLE
[ADF-4488] Form - Add dropdown-cloud widget

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-4488

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+         let optionValue: string = '';
+         if (option.id === 'empty' || option.name !== fieldValue) {
+             optionValue = option.id;
+         } else {
+             optionValue = option.name;
+         }
+         return optionValue;
+     }
+
+    isValidRestType() {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts]: missing return type

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.service.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    contentTypes = ['application/json'];
+    accepts = ['application/json'];
+    returnType = Object;
+
+    constructor(
+        private logService: LogService,
+        private apiService: AlfrescoApiService
+    ) {}
+
+    getDropDownJsonData(url: string) {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.service.ts]: missign return type

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.service.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            this.returnType, null, null)
+        ).pipe(
+            map((res: any) => {
+                return res;
+            }),
+            catchError((err) => this.handleError(err))
+        );
+    }
+
+    private handleError(error: any) {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.service.ts]: missign return type

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                    this.field.options = this.mapJsonData(result);
+                } else {
+                    this.field.options = result;
+                }
+            },
+            (err) => this.handleError(err));
+         }
+     }
+
+    mapJsonData(data: object[]): FormFieldOption[] {
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts]: are we using any in those situations?

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            return {
+                name: value[path][idProperty],
+                id: value.id
+            };
+        });
+     }
+
+    getValuesByProcessDefinitionId() {
+         this.formService
+            .getRestFieldValuesByProcessId(
```
Comment (monsterflaw) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts]: Don't we need a new service for APS cloud?

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+         if (this.field && this.field.restUrl) {
+             if (this.field.form.taskId) {
+                 this.getValuesByTaskId();
+             } else {
+                 this.getValuesByProcessDefinitionId();
+             }
+         }
+     }
+
+    getValuesByTaskId() {
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts]: can we have a meaningful name?

File: lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+            return {
+                name: value[path][idProperty],
+                id: value.id
+            };
+        });
+     }
+
+    getValuesByProcessDefinitionId() {
+         this.formService
+            .getRestFieldValuesByProcessId(
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/form/components/dropdown-cloud/dropdown-cloud.widget.ts]: This service is for APS1? I guess we need a different service

**Pull request merged**